### ISSUE TITLE
Noisy logs

### ISSUE THREAD

**Issue by wngr** [libp2p/rust-libp2p#2088]
We have been trying to get a good grip on logging without code base. Libraries usually are quite silent (as long as they work :-) ), but some modules within rust-libp2p are actually quite noisy. One noteworthy example is `gossipsub`, where the log levels could be tuned down quite a bit:
```
./protocols/gossipsub/src/behaviour.rs
512:        info!("Subscribed to topic: {}", topic);
552:        info!("Unsubscribed from topic: {:?}", topic_hash);
703:        info!("Published message: {:?}", &msg_id);
863:            info!("JOIN: The topic is already in the mesh, ignoring JOIN");
934:            info!("JOIN: Sending Graft message to peer: {:?}", peer_id);
1023:                info!("LEAVE: Sending PRUNE to peer: {:?}", peer);
1319:                    info!(
1357:            info!(
1391:                info!(
1830:                        info!(
2818:        info!("New peer connected: {}", peer_id);

./protocols/gossipsub/src/subscription_filter.rs
69:                info!("Filtered incoming subscription {:?}", s);
```

The events logged on warn level might not warrant a warn in every case either.

I wonder whether it would be acceptable to go through the code base with a fine hair comb and assert the various log levels, or library consumers should selectively configure the log levels for the different modules.

**Comment by mxinden** [comment]
> I wonder whether it would be acceptable to go through the code base with a fine hair comb and assert the various log levels, or library consumers should selectively configure the log levels for the different modules.

I am very much in favor of consistent log-level usage across all crates in rust-libp2p. Having to configure different log levels for each crate should not be needed for most deployments.

That said, I have not found the time / priority to go through the code-case to enforce log-level consistency. @wngr do you have some time to (partially) tackle this? I am happy to do the reviews. In addition I would guess @AgeManning would appreciate the changes in `libp2p-gossipsub`.

**Comment by AgeManning** [comment]
Oh. The timing on this is perfect. I've actually just made these changes because I've also found the gossipsub logs very noisy. 

I'm in the process of tracking down a known issue in the current gossipsub, which has prompted this. 

I've already downgraded some logs here: https://github.com/sigp/rust-libp2p/tree/gossipsub-fix if you want an immediate fix, however once I've tracked down the issue, I intend to make a PR which corrects these logs along with the issue (there is an edge case where gossipsub can currently panic).
